pkg/kubeinteraction/resultsannotation.go

```diff
@@ -0,0 +1,41 @@
+package kubeinteraction
+
+import (
+	"encoding/json"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+const (
+	resGroupName            = "results.tekton.dev"
```

Comment:
may be we can move this to the 
`"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/keys"`
and reuse from there

Reply:
Hey @sm43 👋🏼

> may be we can move this to the
"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/keys"
and reuse from there
 
I can do that. 
But following  [Chmouel's comment](https://github.com/openshift-pipelines/pipelines-as-code/pull/1245#discussion_r1432431483) where he mentions that it's an internal implementation and indeed that is the case, the usage of `resGroupName` is specific to this implementation, and its applicability elsewhere is not immediately clear.
wdyt?

Reply:
I guess that comment is focused on user docs.. moving this annotations to keys is more of keeping those at one place.

Reply:
yeah agreed with @sm43, we centralise all annotations in there to make it easy to refer from everywhere...

Reply:
I've made the suggested changes.